renderlib/StringUtil.cpp

```diff
@@ -0,0 +1,74 @@
+#include "StringUtil.h"
+
+#include "Logging.h"
+
+#include <sstream>
+
+std::string
+trim(const std::string& str, const std::string& whitespace)
+{
+  const auto strBegin = str.find_first_not_of(whitespace);
+  if (strBegin == std::string::npos)
+    return ""; // no content
+
+  const auto strEnd = str.find_last_not_of(whitespace);
+  const auto strRange = strEnd - strBegin + 1;
+
+  return str.substr(strBegin, strRange);
+}
+
+bool
+startsWith(const std::string mainStr, const std::string toMatch)
+{
+  // std::string::find returns 0 if toMatch is found at starting
+  if (mainStr.find(toMatch) == 0)
+    return true;
+  else
+    return false;
+}
+
+bool
+endsWith(std::string const& value, std::string const& ending)
+{
+  if (ending.size() > value.size())
+    return false;
+  return std::equal(ending.rbegin(), ending.rend(), value.rbegin());
```

Comment:
Instead of searching at `value.rbegin()`, should this start at an offset `ending.size` characters from the end of `value`?

Something like
`return std::equal(ending.rbegin(), ending.rend(), value.rend() - (ending.rend() - ending.rbegin()))`